lib/getWorkspacesYarn.js

```diff
@@ -23,21 +23,19 @@ function getWorkspacesYarn(cwd, ignorePackages = null) {
 		throw new TypeError("package.json: workspaces or workspaces.packages: Must be non-empty array of string");
 	}
 
-	// Ignore packages via CLI or via manifest workspaces
-	const _ignorePackages = ignorePackages
-		? ignorePackages
-		: packages.filter((p) => /^!/.test(p)).map((p) => p.replace("!", ""));
-
-	// Remove ignoredPackages from packages
-	if (_ignorePackages.length > 1) packages = packages.filter((p) => !/^!/.test(p));
+	// If packages to be ignored come from CLI, we need to remove the ones from manifest and add the ones from ignorePackages
+	if (ignorePackages && Array.isArray(ignorePackages)) {
+		packages = packages.filter((p) => !/^!/.test(p));
+		packages.push(...ignorePackages.map((p) => `!${p}`));
+	}
```

Comment:
@antongolub still had to maintain this block, but instead of using a `_ignorePackages` inside the glob process, I'm pushing the ignored packages that comes from CLI and pushing them inside the packages with the `!` operator.

Reply:
@davikawasaki, 
 
Why do we need to filter the packages that we get from `package.json.workspaces`? Why not just:
```js
if (Array.isArray(ignorePackages)) {
   packages.push(...ignorePackages.map(p => `!${p}`))
}
```

Reply:
@antongolub I was thinking that the CLI would **override** the options we get from `package.json.workspaces`. Do you think it's best to combine them?

Reply:
@antongolub another reason why I prefer to **override** instead of appending the ignored packages:

![image](https://user-images.githubusercontent.com/12398127/106839186-7764ba80-667c-11eb-8017-f432eefd4e73.png)

Reply:
I think it would not be a huge overhead for globby ). `gitignore` handling impact is much bigger.

Reply:
@antongolub fair enough... I'll make the corrections and add some extra info in the docs about this.